src/app/data/oasis-account-names.ts

```diff
@@ -0,0 +1,120 @@
+import { useQuery } from '@tanstack/react-query'
+import { Layer } from '../../oasis-nexus/api'
+import { Network } from '../../types/network'
+import { hasTextMatch } from '../components/HighlightedText/text-matching'
+import * as process from 'process'
+import { AccountData, AccountEntry, AccountMap, AccountMetadata, AccountMetadataInfo } from './named-accounts'
+
+const dataSources: Record<Network, Partial<Record<Layer, string>>> = {
+  [Network.mainnet]: {
+    [Layer.consensus]:
+      'https://raw.githubusercontent.com/oasisprotocol/nexus/main/account-names/mainnet_consensus.json',
+    [Layer.emerald]:
+      'https://raw.githubusercontent.com/oasisprotocol/nexus/main/account-names/mainnet_paratime.json',
+    [Layer.sapphire]:
+      'https://raw.githubusercontent.com/oasisprotocol/nexus/main/account-names/mainnet_paratime.json',
+  },
+  [Network.testnet]: {
+    [Layer.consensus]:
+      'https://raw.githubusercontent.com/oasisprotocol/nexus/main/account-names/testnet_consensus.json',
+    [Layer.emerald]:
+      'https://raw.githubusercontent.com/oasisprotocol/nexus/main/account-names/testnet_paratime.json',
+    [Layer.sapphire]:
+      'https://raw.githubusercontent.com/oasisprotocol/nexus/main/account-names/testnet_paratime.json',
+  },
+}
+
+const getOasisAccountsMetadata = (network: Network, layer: Layer) =>
+  new Promise<AccountData>((resolve, reject) => {
+    const url = dataSources[network][layer]
+    if (!url) {
+      reject('No data available for this layer')
+    } else {
+      axios.get(url).then(response => {
+        if (response.status !== 200) reject("Couldn't load names")
+        if (!response.data) reject("Couldn't load names")
+        // console.log('Response data is', response.data)
+        const map: AccountMap = new Map()
+        const list: AccountEntry[] = []
+        Array.from(response.data).forEach((entry: any) => {
+          const address = entry.Address
+          const metadata: AccountMetadata = {
+            name: entry.Name,
+            description: entry.Description,
+          }
+          // Register the metadata in its native form
+          list.push({
+            address,
+            ...metadata,
+          })
+          map.set(address, metadata)
+        })
+        resolve({
+          map,
+          list,
+        })
+      }, reject)
+    }
+  })
+
+export const useOasisAccountsMetadata = (network: Network, layer: Layer, enabled: boolean) => {
+  return useQuery(['oasisAccounts', network, layer], () => getOasisAccountsMetadata(network, layer), {
+    enabled,
+    staleTime: Infinity,
+  })
+}
+
+export const useOasisAccountMetadata = (
+  network: Network,
+  layer: Layer,
+  address: string,
+  enabled: boolean,
+): AccountMetadataInfo => {
+  // When running jest tests, we don't want to load from Pontus-X.
+  if (process.env.NODE_ENV === 'test') {
+    return {
+      metadata: {
+        name: undefined,
+      },
+      loading: false,
+    }
+  }
```

Comment:
This would have been better if it was mocked in tests(`useOasisAccountMetadata` function), instead of being bundled to production code.

Reply:
OK, removed mocking from production code.